tests/component/test_task_events.py

```diff
@@ -55,6 +56,118 @@ def test___done_event_registered___run_finite_acquisition___callback_invoked_onc
     assert all(e.status == 0 for e in event_observer.events)
 
 
+@pytest.mark.grpc_xfail(
+    reason="#559: Can't close tasks in an event callback when using gRPC", raises=AssertionError
+)
+def test___done_event_registered___call_clear_in_callback___stop_close_in_callback_with_success_status(
+    sim_6363_device: nidaqmx.system.Device,
+    init_kwargs: dict,
+) -> None:
+
+    # We have to define and set this here because mypy will error if it is
+    # defined below with everything else: https://github.com/python/mypy/issues/7057
+    side_effect_semaphore = threading.Semaphore(value=0)
+
+    # side_effect for callback
+    def _clear_task():
+        task.stop()
+        task.close()
+        side_effect_semaphore.release()
+
+    try:
+        # We need to create our own task here because we need to call stop() and close() on it.
+        task: nidaqmx.Task = nidaqmx.Task(**init_kwargs)
+        task.ai_channels.add_ai_voltage_chan(
+            sim_6363_device.ai_physical_chans[0].name, max_val=10, min_val=-10
+        )
+    except nidaqmx.DaqError:
+        if task is not None:
+            task.close()
+        raise
+    event_observer = DoneEventObserver(side_effect=_clear_task)
+    task.register_done_event(event_observer.handle_done_event)
+    task.timing.cfg_samp_clk_timing(
+        rate=10000.0, sample_mode=AcquisitionType.FINITE, samps_per_chan=1000
+    )
+
+    task.start()
+    event_observer.wait_for_events(1)
+    acquired = side_effect_semaphore.acquire(timeout=1.0)
+    # Ensure we get the expected exception and warning
+    with pytest.raises(nidaqmx.DaqError) as exc_info:
+        task.stop()
+    with pytest.warns(
+        DaqResourceWarning,
+        match="Attempted to close NI-DAQmx task",
+    ) as warnings_record:
+        task.close()
```

Comment:
This looks like it's asserting that the task was already closed, which belongs in "assert" not "act"

Reply:
Handled in #564